src/vopr.zig

```diff
@@ -0,0 +1,472 @@
+        .Signal => |code| {
+            switch (code) {
+                6 => {
+                    std.debug.print("exit with signal: {}. Indicates a crash bug.\n", .{code});
+                    return @enumToInt(Bug.crash);
+                },
+                else => {
+                    fatal("the simulator exited with an unexpected signal. Term: {}\n", .{term});
+                },
+            }
+        },
+        else => {
+            fatal("the simulator exited without an exit code. Term: {}\n", .{term});
+        },
+    }
+}
+
+fn check_git_status(allocator: mem.Allocator) void {
+    // Running git status to determine whether there is any uncommitted code or local changes.
+    var args = [2][]const u8{ "git", "status" };
+    var exec_result = std.ChildProcess.exec(.{
+        .allocator = allocator,
+        .argv = &args,
+    }) catch |err| {
+        fatal("unable to determine TigerBeetle's git status. Error: {}", .{err});
+    };
+
+    var git_status = exec_result.stdout;
+    if (mem.containsAtLeast(u8, git_status, 1, "nothing to commit, working tree clean") and
+        mem.containsAtLeast(u8, git_status, 1, "Your branch is up to date"))
+    {
+        std.debug.print("All code has been committed and pushed.\n", .{});
+    } else {
+        fatal(
+            "the VOPR cannot run with --send when your branch is ahead or there's uncommited code",
+            .{},
+        );
+    }
+}
+
+// Sends a bug report to the VOPR Hub.
+// The VOPR Hub will attempt to verify the bug and automatically create a GitHub issue.
+fn send_report(allocator: mem.Allocator, address: net.Address, bug: Bug, seed: u64) void {
+    var message: Report = create_report(allocator, bug, seed);
+    var byte_array: [LENGTH_OF_VOPR_MESSAGE]u8 = undefined;
+    var hash: [32]u8 = undefined;
+
+    // Bug type
+    assert(message.bug == 1 or message.bug == 2 or message.bug == 3);
+
+    // Seed
+    var seed_integer_value: u64 = message.seed;
+    // Zig stores value as Little Endian when VOPR Hub is expecting Big Endian.
+    seed_integer_value = @byteSwap(u64, seed_integer_value);
+    const seed_byte_array: [8]u8 = @bitCast([8]u8, seed_integer_value);
+
+    // Leave the first 16 bytes open to store the first half of a SHA256 hash of the message.
+    byte_array[16] = message.bug;
+    var offset: usize = 17;
+    mem.copy(u8, byte_array[offset..], &seed_byte_array);
+    offset = offset + seed_byte_array.len;
+    mem.copy(u8, byte_array[offset..], &message.commit);
+
+    // Hash the bug report message.
+    std.crypto.hash.sha2.Sha256.hash(byte_array[16..], hash[0..], .{});
+    // Append the first half of the hash to the beginning of the byte array.
+    mem.copy(u8, byte_array[0..16], hash[0..16]);
+
+    // Send message
+    const stream = net.tcpConnectToAddress(address) catch |err| {
+        fatal("unable to create a connection to the VOPR Hub. Error: {}", .{err});
+    };
+
+    std.debug.print("Connected to VOPR Hub.\n", .{});
```

Comment:
Replace all `std.debug.print()`s with `log.debug()`, to be consistent with our other Zig code.